extensions/machine-learning-services/src/test/packageManagement/packageManagementService.test.ts

```diff
@@ -23,74 +23,64 @@ function createContext(): TestContext {
 		const context = createContext();
 		context.queryRunner.setup(x => x.isPythonInstalled(TypeMoq.It.isAny())).returns(() => Promise.resolve(true));
-		let serverConfigManager = new ServerConfigManager(context.apiWrapper.object, context.queryRunner.object);
-		let connection  = new azdata.connection.ConnectionProfile();
+		let serverConfigManager = new PackageManagementService(context.apiWrapper.object, context.queryRunner.object);
+		let connection = new azdata.connection.ConnectionProfile();
 		should.equal(await serverConfigManager.isPythonInstalled(connection), true);
 	});
 
-	it('updateExternalScriptConfig should show info message if updated successfully', async function (): Promise<void> {
-		const context = createContext();
-		context.queryRunner.setup(x => x.updateExternalScriptConfig(TypeMoq.It.isAny(), true)).returns(() => Promise.resolve());
-		context.queryRunner.setup(x => x.isMachineLearningServiceEnabled(TypeMoq.It.isAny())).returns(() => Promise.resolve(true));
-		context.apiWrapper.setup(x => x.showInfoMessage(TypeMoq.It.isAny())).returns(() => Promise.resolve(''));
-		context.apiWrapper.setup(x => x.showErrorMessage(TypeMoq.It.isAny())).returns(() => Promise.resolve(''));
-		let serverConfigManager = new ServerConfigManager(context.apiWrapper.object, context.queryRunner.object);
-		let connection  = new azdata.connection.ConnectionProfile();
-		await serverConfigManager.updateExternalScriptConfig(connection, true);
-
-		context.apiWrapper.verify(x => x.showInfoMessage(TypeMoq.It.isAny()), TypeMoq.Times.once());
-	});
-
-	it('updateExternalScriptConfig should show error message if did not updated successfully', async function (): Promise<void> {
+	it('enableExternalScriptConfig should show error message if did not updated successfully', async function (): Promise<void> {
 		const context = createContext();
 		context.queryRunner.setup(x => x.updateExternalScriptConfig(TypeMoq.It.isAny(), true)).returns(() => Promise.resolve());
 		context.queryRunner.setup(x => x.isMachineLearningServiceEnabled(TypeMoq.It.isAny())).returns(() => Promise.resolve(false));
 		context.apiWrapper.setup(x => x.showInfoMessage(TypeMoq.It.isAny())).returns(() => Promise.resolve(''));
 		context.apiWrapper.setup(x => x.showErrorMessage(TypeMoq.It.isAny())).returns(() => Promise.resolve(''));
-		let serverConfigManager = new ServerConfigManager(context.apiWrapper.object, context.queryRunner.object);
-		let connection  = new azdata.connection.ConnectionProfile();
-		await serverConfigManager.updateExternalScriptConfig(connection, true);
+		context.apiWrapper.setup(x => x.showQuickPick(TypeMoq.It.isAny(), TypeMoq.It.isAny(), TypeMoq.It.isAny())).returns(() => Promise.resolve({
+			label: 'Yes'
+		}));
+		let serverConfigManager = new PackageManagementService(context.apiWrapper.object, context.queryRunner.object);
+		let connection = new azdata.connection.ConnectionProfile();
+		await serverConfigManager.enableExternalScriptConfig(connection);
```

Comment:
going forward we would probably want integration tests to actually do this. please correct me if we already have or dont need that.

Reply:
I have integration test for sqltoolservice and will use sqltoolsservice api in another PR so removing the tsql from here. I don't think more integration tests needed but maybe I'm wrong

---
In reply to: [393911023](https://github.com/microsoft/azuredatastudio/pull/9519#discussion_r393911023) [](ancestors = 393911023)